### ISSUE TITLE
Auto-Create Namespace for Mirrored Services if None Exists

### ISSUE THREAD

**Issue by deusxanima** [linkerd/linkerd2#10912]
### What problem are you trying to solve?

It would be useful to modify the cluster role used by linkerd multicluster service mirror to allow it to auto-create a namespace on the mirrored cluster if one does not already exist. Right now service mirroring is very easy with the annotation setup, however, this only works if the remote namespace matching the source namespace exists. If not then users have to manually add in additional steps into the pipeline to verify and create missing namespaces. It would be very useful if the service mirror could do this check and create the namespaces on its own if a matching ns did not exist.

### How should the problem be solved?

Modifying the existing clusterrole used by the mc service mirror to allow it to create namespaces.

### Any alternatives you've considered?

n/a

### Would you like to work on this feature?

None

**Comment by knutgoetz** [comment]
I am a bit confused.  We ran into the issue that a service could not be mirrored because the namespace did not exist. I do remember that automatically creating the namespace in these cases was the default behavior of the service mirror in the past. It was [changed](https://github.com/linkerd/linkerd2/commit/99b64a601f78fb24c3ee85d68e2157ee3fdbb25f)  due to [issues with managing the namespaces with IaC tooling](https://github.com/linkerd/linkerd2/issues/7565). Are you planning to add this mechanic again? If not, is there a good way to monitor for missing namespaces for mirrored services?

**Comment by ericlanderson** [comment]
Yeah, as referenced in #7565, ideally this would be an opt-in via a flag to the helm chart or CLI.

**Comment by alpeb** [comment]
Implemented in #13137
